chef/cookbooks/neutron/recipes/post_install_conf.rb

```diff
@@ -61,26 +62,33 @@ def mask_to_bits(mask)
 end
 neutron_cmd = "neutron #{neutron_args}"
 
+fixed_network_type = ""
 floating_network_type = ""
-if node[:neutron][:networking_mode] == 'vlan'
-  fixed_network_type = "--provider:network_type vlan --provider:segmentation_id #{fixed_net["vlan"]} --provider:physical_network physnet1"
-  if node[:network][:networks][:nova_floating][:use_vlan]
-    floating_network_type = "--provider:network_type vlan --provider:segmentation_id #{floating_net["vlan"]} --provider:physical_network physnet1"
-  else
+
+networking_plugin = node[:neutron][:networking_plugin]
+ml2_type_drivers_default_provider_network = node[:neutron][:ml2_type_drivers_default_provider_network]
+case networking_plugin
+when 'ml2'
+  case ml2_type_drivers_default_provider_network
+  when 'vlan'
+    fixed_network_type = "--provider:network_type vlan --provider:segmentation_id #{fixed_net["vlan"]} --provider:physical_network physnet1"
     floating_network_type = "--provider:network_type flat --provider:physical_network physnet1"
```

Comment:
> But actually there might be another issue here. We lost the:
> 
> if node[:network][:networks][:nova_floating][:use_vlan]
>     floating_network_type = "......."
> thing. Which makes we wonder why floating IPs still worked yesterday ... I'll recheck.

As discussed with @rhafer I readded the current behavior in https://github.com/crowbar/barclamp-neutron/commit/ccf35be763d9761c70c757f1b8d9789de4630d03